src/test/unit/algorithms/ConnectionsPerformanceTest.cpp

```diff
@@ -59,33 +56,31 @@ float runTemporalMemoryTest(UInt numColumns, UInt w,   int numSequences,
   // Initialize
 
   TemporalMemory tm;
-  vector<UInt> columnDim;
-  columnDim.push_back(numColumns);
-  tm.initialize(columnDim);
+  tm.initialize( {numColumns} );
 
   cout << (float)timer.getElapsed() << " in " << label << ": initialize"  << endl;
 
   // Learn
 
-  vector<vector<vector<CellIdx>>> sequences;
-  vector<vector<CellIdx>> sequence;
-  vector<CellIdx> sdr;
-
+  vector<vector<SDR>> sequences;
   for (int i = 0; i < numSequences; i++) {
+    vector<SDR> sequence;
+    SDR sdr({numColumns});
     for (int j = 0; j < numElements; j++) {
-      sdr = _randomSDR(numColumns, w);
+      const Real sparsity = w / static_cast<Real>(numColumns);
+      sdr.randomize(sparsity, rng);
       sequence.push_back(sdr);
     }
-
     sequences.push_back(sequence);
   }
 
+  const SDR unused_({std::numeric_limits<UInt>::max()}); //dummy SDR, unused here, but needed for API
```

Comment:
Note: for the external predictive input arguments, if they're not being used then any SDR will do, it doesn't need to be this big.

Off Topic: The TM should probably have a convenience method which doesn't require these or has defaults.

Reply:
> Off Topic: The TM should probably have a convenience method which doesn't require these or has defaults. 

yes, I'll add it in a PR that does away with "old compute" 

> any SDR will do, it doesn't need to be this big. 

oh, thanks. My bad, I'm always confusing this with an initializer-list for values!

Reply:
I rather added the convenience TM::compute(SDR), so no helper SDR is needed.